arcs/src/algorithms/scale.rs

```diff
@@ -1,119 +1,158 @@
 use crate::{
-    primitives::{Arc, Line, Point},
-    Vector,
+    primitives::{Arc},
+    algorithms::ScaleNonUniform,
+    components::{BoundingBox},
 };
-use kurbo::Affine;
 
 /// Something which can be scaled in *Drawing Space*
 pub trait Scale {
-    fn scale(&mut self, factor: f64, base: Vector);
+    fn scale(&mut self, scale_factor: f64);
 
-    fn scaled(&self, factor: f64, base: Vector) -> Self
+    fn scaled(&self, scale_factor: f64) -> Self
     where 
         Self: Sized + Clone,
         {
             let mut clone = self.clone();
-            clone.scale(factor, base);
+            clone.scale(scale_factor);
 
             clone
         }
 
 }
 
-impl<'t, T: Scale + ?Sized> Scale for &'t mut T {
-    fn scale(&mut self, factor: f64, base: Vector) {
-        (*self).scale(factor, base);
-    }
-}
-
-impl Scale for Vector {
-    fn scale(&mut self, factor: f64, base: Vector) {
-        let translate_to_base = Affine::translate(base * -1.0);
-        let scale = Affine::scale(factor);
-        let translate_back = Affine::translate(base);
-        let combined_transform = translate_back * scale * translate_to_base;
-        let new_pos = combined_transform * *self;
-        self.x = new_pos.x;
-        self.y = new_pos.y;
-    }
-}
-
-impl Scale for Point {
-    fn scale(&mut self, factor: f64, base: Vector) {
-        self.location.scale(factor, base);
-    }
-}
-
-impl Scale for Line {
-    fn scale(&mut self, factor: f64, base: Vector) {
-        self.start.scale(factor, base);
-        self.end.scale(factor, base);
+impl<S: ScaleNonUniform> Scale for S {
+    fn scale(&mut self, scale_factor: f64) {
+        self.scale_non_uniform(scale_factor, scale_factor);
     }
 }
 
 impl Scale for Arc {
-    fn scale(&mut self, factor: f64, base: Vector) {
-        let mut centre = self.centre();
-        centre.scale(factor, base);
+    fn scale(&mut self, scale_factor: f64) {
         *self = Arc::from_centre_radius(
-            centre, 
-            self.radius() * factor, 
+            self.centre().scaled(scale_factor), 
+            self.radius() * scale_factor, 
             self.start_angle(), 
             self.sweep_angle(),
         );   
     }
 }
 
+impl Scale for BoundingBox {
```

Comment:
We can probably get this for free by implementing `ScaleNonUniform` for `BoundingBox`.

Reply:
Yes makes a lot more sense!

implemented here: https://github.com/DerLando/arcs/tree/implement-affinetransformable-for-scale-and-translate

Reply:
@DerLando, I can't see the changes you're talking about. Have they been pushed to GitHub?